packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1002,6 +1002,112 @@ class _SliverAppBarDelegate extends SliverPersistentHeaderDelegate {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=freeform}
+///
+/// ```dart imports
+/// import 'package:flutter/material.dart';
+/// ```
+///
+/// ```dart
+/// void main() => runApp(MyApp());
+///
+/// class MyApp extends StatefulWidget {
+///   const MyApp({Key key}) : super(key: key);
+///
+///   @override
+///   State<StatefulWidget> createState() => _MyAppState();
+/// }
+/// class _MyAppState extends State<MyApp> {
+///   bool _pinned = true;
+///   bool _snap = false;
+///   bool _floating = false;
+///
+///   // SliverAppBar is declared in Scaffold.body, in slivers of a
+///   // CustomScrollView.
+///   @override
+///   Widget build(BuildContext context) {
+///     return MaterialApp(
+///       home: Scaffold(
+///         body: CustomScrollView(
+///           slivers: <Widget>[
+///             SliverAppBar(
+///               pinned: this._pinned,
+///               snap: this._snap,
+///               floating: this._floating,
+///               expandedHeight: 160.0,
+///               flexibleSpace: FlexibleSpaceBar(
+///                 title: const Text("SliverAppBar"),
+///                 background: FlutterLogo(),
+///               ),
+///             ),
+///             SliverFillRemaining(
+///               child: Center(
+///                   child:
+///                       Text("Scroll to see SliverAppBar in effect .")),
+///             ),
```

Comment:
The indentation here is off. Also, let's add the `hadScrollBody: false,` to the `SliverFillRemaining`.

Reply:
If we add hadScrollBody: false to the SliverFillRemaining , then user won't be able to scroll to see SliverApppbar in effect .

Reply:
Oh I see. It might be better to use a different sliver, like a SliverToBoxAdapter with a really large container in it. While SliverFillRemaining creates a desirable effect here to show off the SliverAppBar behaviors, it can be kind of confusing to see it used out of the context someone might usually use it for.

Reply:
I've added SliverToBoxAdapter ,let me know if anything else needs to be changed .